src/main/java/org/testng/internal/RuntimeBehavior.java

```diff
@@ -31,4 +33,13 @@ public static TimeZone getTimeZone() {
         }
         return TimeZone.getTimeZone(timeZone);
     }
+
+  /**
+   * @return - <code>true</code> if we would like to invoke applicable listeners for tests that
+   *     would be skipped by TestNG due to a configuration failure (or) due to a upstream dependency
+   *     failure.
+   */
+  public static boolean invokeListenersForSkippedTests() {
+    return Boolean.parseBoolean(System.getProperty(TESTNG_LISTENERS_ALWAYSRUN, "false"));
```

Comment:
So the default behavior is to omit notifications for skipped tests. This diverges from the behavior of previous releases. By exhibiting altered behavior by default, you require folks who have implemented their ITestListener classes based on the previous behavior to discover and set a system property to avoid being broken when they upgrade to the latest release.

Reply:
The behavior changed since `6.9.11`. So, what do you mean by "previous releases"?
If we rollback to behavior before `6.9.11`, we will have a regression for users which are expecting the new behavior.

BTW, if you have a better solution, feel free to propose a pull request and we will do our best to merge it if it is possible.

Reply:
Are you sure about the behavior change? I'm running with TestNG 6.10, and it sends all expected notifications. (I never upgraded past 6.10 due to other unresolved listener issues.)